### PR TITLE
Pre Publish CPT function and filter.

### PR DESCRIPTION
**Pull request by Rahullakhwani**
Added CPT functionality and filter options.

### CODE REVIEW

File: classes/class-ppc-loader.php [hunk 1]
```diff
@@ -47,33 +47,46 @@ public static function get_instance() {
...
-				'ppc_key4' => 'Formatting Done',
-				'ppc_key5' => 'Title is Catchy',
-				'ppc_key6' => 'Social Images Assigned',
-				'ppc_key7' => 'Done SEO',
-				'ppc_key8' => 'Spelling and Grammar Checked',
-			);
-			$ppc_default_post_types     = array( 'post', 'page' );
-			add_option( 'ppc_checklist_data', $ppc_default_checklist_data );
-			add_option( 'ppc_post_types_to_display', $ppc_default_post_types );
+			add_option( 'ppc_post_types_to_display', array( 'post', 'page' ) );
```
Comment (patilvikasj) [inline classes/class-ppc-loader.php]: This add_option will load on every load. If this is a static array. Declare this as a static variable.

File: classes/class-ppc-pagesetups.php [hunk 2]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+					} elseif ( isset( $ppc_post_val_array ) && empty( $ppc_checked_data ) ) {
+						echo esc_html( sprintf( __( 'Checklist is empty', 'pre-publish-checklist' ) ) );
+					} elseif ( isset( $ppc_post_val_array ) && ! empty( $ppc_post_val_array ) && isset( $ppc_checked_data ) && ! empty( $ppc_checked_data ) ) {
+						$ppc_result = array_intersect_key( $ppc_post_val_array, $ppc_checked_data );
+						if ( isset( $ppc_result ) && ! empty( $ppc_result ) && count( $ppc_post_val_array ) > count( $ppc_result ) ) {
+							/* translators: %d: number term */
+							echo esc_html( sprintf( __( '%1$d items completed out of %2$d', 'pre-publish-checklist' ), count( $ppc_result ), count( $ppc_post_val_array ) ) );
+							echo '<br>';
+							?>
+						<progress value="<?php echo (int) count( $ppc_result ); ?>" max="<?php echo (int) count( $ppc_post_val_array ); ?>"></progress>
```
Comment (patilvikasj) [inline classes/class-ppc-pagesetups.php]: No need to typecast result from count function. It returns the integer number.

File: classes/class-ppc-pagesetups.php [hunk 3]
```diff
@@ -52,11 +52,173 @@ public static function get_instance() {
...
+			$ppc_pst_type       = get_option( 'ppc_post_types_to_display' );
+			$ppc_checklist_data = PPC_Loader::get_instance()->get_list();
+			if ( isset( $ppc_pst_type ) && isset( $ppc_checklist_data ) ) {
+				foreach ( $ppc_pst_type as $key ) {
+					$ppc_page_data     = isset( $ppc_checklist_data[ $key ] ) ? $ppc_checklist_data[ $key ] : '';
+					$current_post_type = get_current_screen();
+					if ( $current_post_type->post_type === $key ) {
+						?>
+						<select class="select_multiple" name="slect_opt" id="slect_opt">
+							<option value="">Filter Unchecked...</option>
```
Comment (patilvikasj) [inline classes/class-ppc-pagesetups.php]: The text should be always ready to translate. here _Filter Unchecked_ is not translatable.

File: classes/class-ppc-update.php [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+ * PHP version 7
+ *
+ * @category PHP
+ * @package  Pre-Publish Checklist.
+ * @author   Display Name <username@brainstormforce.com>
+ * @license  http://brainstormforce.com
+ * @link     http://brainstormforce.com
+ */
+
+if ( ! class_exists( 'PPC_Update' ) ) :
```
Comment (patilvikasj) [inline classes/class-ppc-update.php]: Remove class_exists check. This class already has a prefix. Also, this is defined from plugin itself. We are not using a third party plugin class.

File: classes/class-ppc-update.php [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+		}
+
+		/**
+		 * Moves the option values from ppc_checklist_data to ppc_cpt_checklist_data.
+		 *
+		 * @since 1.1.0
+		 */
+		public function update_options_table() {
+			$ppc_default_checklist_data = array(
+				'ppc_key2' => 'Featured Image Assigned',
```
Comment (patilvikasj) [inline classes/class-ppc-update.php]: Strings need to be translation ready here too.

File: includes/ppc-frontend.php [hunk 6]
```diff
@@ -5,16 +5,18 @@
  *
```
Comment (patilvikasj) [inline includes/ppc-frontend.php]: Prevent direct access to single PHP files as this one. [https://www.barrykooij.com/prevent-direct-file-access-plugin-files/](url)

File: includes/ppc-tabs.php [hunk 7]
```diff
@@ -45,10 +45,14 @@
...
-
-		<a href="?page=ppc&tab=ppc-checklist" class="nav-tab tb 
+	">
+	<?php esc_attr_e( 'General Settings', 'pre-publish-checklist' ); ?></a>
+		<?php
+		$ppc_post_types       = get_option( 'ppc_post_types_to_display' );
+		$ppc_type             = isset( $ppc_post_types[0] ) ? $ppc_post_types[0] : '';
+		$ppc_active_checkpost = ! empty( $ppc_post_types ) ? $ppc_type : 'post';
+		?>
+		<a href="?page=ppc&tab=ppc-checklist&type=<?php echo esc_attr( $ppc_active_checkpost ); ?>" class="nav-tab tb
```
Comment (patilvikasj) [inline includes/ppc-tabs.php]: Using add_query_arg for anchor link would be better here. [https://developer.wordpress.org/reference/functions/add_query_arg/](url)

File: classes/class-ppc-update.php (hunk 4 above)
Comment (Rahullakhwani) [reply classes/class-ppc-update.php]: class_exists removed.

File: includes/ppc-frontend.php (hunk 6 above)
Comment (Rahullakhwani) [reply includes/ppc-frontend.php]: added 
if ( ! defined( 'ABSPATH' ) ) {
	exit; // Exit if accessed directly.
}
at the start of ppc-frontend.php file.

File: classes/class-ppc-pagesetups.php (hunk 3 above)
Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: <option value=""><?php esc_html_e( 'Filter Unchecked...', 'pre-publish-checklist' ); ?></option>

Comment (Rahullakhwani) [reply classes/class-ppc-pagesetups.php]: Added <?php esc_html_e( 'Filter Unchecked...', 'pre-publish-checklist' ); ?>

File: classes/class-ppc-update.php (hunk 5 above)
Comment (Rahullakhwani) [reply classes/class-ppc-update.php]: __('Featured Image Assigned', 'pre-publish-checklist' ),

Added for full checklist.

File: includes/ppc-tabs.php (hunk 7 above)
Comment (Rahullakhwani) [reply includes/ppc-tabs.php]: Used  add_query_arg

**Pull request merged**